src/fcmp_pp/curve_trees.cpp

```diff
@@ -65,49 +65,54 @@ template Selene::Point get_new_parent<Selene>(const std::unique_ptr<Selene> &cur
 template Helios::Point get_new_parent<Helios>(const std::unique_ptr<Helios> &curve,
     const typename Helios::Chunk &new_children);
 //----------------------------------------------------------------------------------------------------------------------
-OutputTuple output_to_tuple(const OutputPair &output_pair)
+OutputTuple output_to_tuple(const OutputPair &output_pair, bool torsion_checked, bool use_fast_check)
 {
     const crypto::public_key &output_pubkey = output_pair.output_pubkey;
     const rct::key &commitment              = output_pair.commitment;
 
     const rct::key &O_key = rct::pk2rct(output_pubkey);
     const rct::key &C_key = commitment;
 
-    TIME_MEASURE_NS_START(ge_frombytes_vartime_ns);
-
-    ge_p3 O_p3, C_p3;
-    if (ge_frombytes_vartime(&O_p3, O_key.bytes) != 0)
-        throw std::runtime_error("output pubkey is invalid");
-    if (ge_frombytes_vartime(&C_p3, C_key.bytes) != 0)
-        throw std::runtime_error("commitment is invalid");
-
-    TIME_MEASURE_NS_FINISH(ge_frombytes_vartime_ns);
-
-    TIME_MEASURE_NS_START(identity_check_ns);
-
-    if (fcmp_pp::mul8_is_identity(O_p3))
-        throw std::runtime_error("O mul8 cannot equal identity");
-    if (fcmp_pp::mul8_is_identity(C_p3))
-        throw std::runtime_error("C mul8 cannot equal identity");
-
-    TIME_MEASURE_NS_FINISH(identity_check_ns);
-
-    TIME_MEASURE_NS_START(check_torsion_ns);
-
-    const bool O_is_torsion_free = fcmp_pp::torsion_check_vartime(O_p3);
-    const bool C_is_torsion_free = fcmp_pp::torsion_check_vartime(C_p3);
+    TIME_MEASURE_NS_START(clear_torsion_ns);
```

Comment:
Personally, I would put this declaration in the `if (!torsion_checked)` block so the timer only times when torsion is actually checked.